pkg/util/util.go

```diff
@@ -247,7 +259,12 @@ func GetDNS1123Name(str string) string {
 		":", "-",
 		"--", "-",
 	)
-	return strings.TrimSpace(replacer.Replace(strings.ToLower(str)))
+	withReplacedChars := replacer.Replace(strings.ToLower(str))
+
+	regex := regexp.MustCompile("^[^a-zA-Z0-9]+(.*)$") //regex that strips all leading non alpha-numeric chars
```

Comment:
Don't we need to take care of the trailing non alpha-numeric chars?

Reply:
If you get a reference that they shouldn't be there, I will fix it :)

Reply:
I named a folder "/nodejs-ex#@@@__5" and it failed
```
[mrinaldas@localhost odo]$ odo create nodejs --local /home/mrinaldas/nodejs-ex#@@@__5/
 ✓  Checking component
 ✓  Checking component version
nodejs-ex#@@-nodejs-qnva is not a valid name:  a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
```

Reply:
Awesome, thanks!

Reply:
I know, I do crazy stuffs, most of the times :D

Reply:
BTW can you reproduce this?